### PR TITLE
Fix IncludeNuGetResolvedAssets to handle packages in nuget fallback directory correctly

### PR DESCRIPTION
**Pull request by tmat**
Infrastructure only change.

The previous implementation assumed that all packages are restored under NuGetPackageRoot. That's not necessarily true. Platform packages may be stored under NuGet fallback directory.

### CODE REVIEW

**Comment by tmat** [comment]
@jasonmalinowski @jaredpar @dotnet/roslyn-infrastructure Please review.

File: build/scripts/build.ps1 [hunk 1]
```diff
@@ -27,7 +27,6 @@ param (
     [switch]$sign = $false,
     [switch]$pack = $false,
     [switch]$packAll = $false,
-    [switch]$binaryLog = $false,
```
Comment (jaredpar) [inline build/scripts/build.ps1]: Why are you deleting this option? There is a perf difference when building with / without this option. This is why it wsa left an option.

**Comment by jaredpar** [comment]
> 'm not sure why it was moved to a file-path filtering approach, but I'm guessing @jaredpar either did that ...

Commit that used the file-path approach is https://github.com/dotnet/roslyn/commit/781a9fd8b1d3928f001eb374638a392ee76281c0

The commit message has the details:

> The NuGetPackageToIncludeInVsix element is used to make the NuGet assets
we use in a VSIX very explicit. It depended on the nuget items having
the metadata element NuGetPackageId. The 1.X New SDK no longer has this
piece of metadata and hence we had to pick a new approach.

Basically the 1.X SDK broke us here. Guessing the 2.X SDK added back this metadata. Should have added a bug link in the code :frowning:

File: build/scripts/build.ps1 (hunk 1 above)
Comment (jaredpar) [reply build/scripts/build.ps1]: Okay

Comment (tmat) [reply build/scripts/build.ps1]: You are right. Binary log slows down Roslyn build by 1 minute (4:15 vs 3:15). I'll make it opt-in again.

**Pull request merged**